connection.go

```diff
@@ -391,6 +393,13 @@ func Connect(addr string, opts Opts) (conn *Connection, err error) {
 		}
 	}
 
+	if err = conn.loadProtocolInfo(); err != nil {
+		conn.mutex.Lock()
+		defer conn.mutex.Unlock()
+		conn.closeConnection(err, true)
+		return nil, err
+	}
```

Comment:
It will be a better place for it:

https://github.com/tarantool/go-tarantool/blob/0081afbcec293ca5044de86164714764a0052915/connection.go#L513

Reply:
Yeah, that's a nice one. Moved